### DISCUSSION TITLE
Bug found using the getApplicableInputs function

### DISCUSSION THREAD

**Discussion by ErickRomeroDev** [input-output-hk/marlowe-ts-sdk#186] [Q&A]
Hi Marlowe Team, i was testing the SDK and doing some examples. I found a bug when using the function getApplicableInputs.

I am attaching the source code where the bug was found and how i managed to temporarily solve it.
GITHUB: https://github.com/Erickrs2/marlowe-sdk.git
Error found at: marlowe-sdk/app/buy-coffee/_components/coffees-to-funds.tsx  line 118

problem: The response of the function is swapping the value of party and into_account.

**Comment by hrajchert** [comment]
Thanks for the report 🙌.

We are refactoring the applicable inputs feature to have a better API and to work seamless with Merkleized contracts. This work should be part of the 0.4.0 release.

If you want to peek how this will look like, there is an [experimental-feature folder](https://github.com/input-output-hk/marlowe-ts-sdk/blob/cb546a599d553aa2171a5dcd324eee21bbeea2f9/examples/nodejs/src/experimental-features/applicable-inputs.ts) with the new code. And you can [see here](https://github.com/input-output-hk/marlowe-ts-sdk/blob/cb546a599d553aa2171a5dcd324eee21bbeea2f9/examples/nodejs/src/marlowe-object-flow.ts#L269-L337) how it is being used.

**Reply by ErickRomeroDev** [reply]
excelent, i will take a look at the experimental folder. Any date for release? in case there is one?

**Reply by hrajchert** [reply]
There isn't a date yet, but this is the current feature I'm working on

**Reply by hrajchert** [reply]
Hi @Erickrs2 , I've just made the PR for this, feel free to add any comments or suggestions https://github.com/input-output-hk/marlowe-ts-sdk/pull/187
